packages/catalog-search/src/SearchBox.jsx

```diff
@@ -54,40 +59,61 @@ export const SearchBoxBase = ({
     }
   };
 
+  const handleSuggestionSubmit = (query, hit) => {
+    // If an override is provided, call it with the hit
+    if (suggestionSubmitOverride !== undefined) {
+      suggestionSubmitOverride(hit);
+    }
+
+    // When a suggested search is clicked, set the course tile to the search query
+    setSearchQuery(query);
+    // Do all the regular submit stuff
+    handleSubmit(query);
+  };
+
   useEffect(() => {
     document.addEventListener('click', () => setShowSuggestions(false));
   });
 
+  // Track the focused element
+  const focusedElement = useActiveElement();
+
   useEffect(() => {
     let isMounted = true;
-    const fetchCourses = setTimeout(async () => {
-      if (searchQuery !== '' && isMounted) {
-        const { hits, nbHits } = await index.search(searchQuery, {
-          filters,
-          attributesToHighlight: ['title'],
-          attributesToRetrieve: [
-            'key',
-            'content_type',
-            'title',
-            'authoring_organizations',
-            'aggregation_key',
-            '_highlightResult',
-            'program_type',
-          ],
-        });
-        if (nbHits > 0 && isMounted) {
-          setAutocompleteHits(hits);
-          setShowSuggestions(true);
+
+    // If the component is provided an index and the current focused element is the input field,
+    // start the suggestion request to algolia
+    if (index !== undefined && focusedElement.classList.contains(SEARCH_BOX_CLASS_NAME)) {
```

Comment:
Similar question. Could `index` ever be `null` here, or if `null` is provided, does the default prop take precedence?

Reply:
similar situation, no `index` prop will force it to undefined. You could provide null through the prop to break things but you could also do that by providing any other junk value to the prop